lib/ruby_lsp/check_docs.rb

```diff
@@ -86,7 +94,7 @@ def run_task
             # end
             # ```
           DOCS
-        elsif !/\[.* demo\]\(.*\.gif\)/.match?(documentation)
+        elsif !/\[.* demo\]\(.*\.gif\)/.match?(documentation) || !gif_exists?(file_path)
```

Comment:
I think the error message on this branch is more tailored to the comment missing the `[]()` gif link. Can we add a different branch for this case saying something like "The GIF referenced in the documentation does not exist"?